lib/exercises.rb

```diff
@@ -22,8 +58,81 @@ def top_k_frequent_elements(list, k)
 #   Each element can either be a ".", or a digit 1-9
 #   The same digit cannot appear twice or more in the same 
 #   row, column or 3x3 subgrid
-# Time Complexity: ?
-# Space Complexity: ?
+# Reference: https://www.tutorialspoint.com/valid-sudoku-in-python
+# Time Complexity: O(n^2) where n is the length of the input table and since table is a 2d array
+# Space Complexity: O(1) where the extra hashes are constant tracking 1~9
 def valid_sudoku(table)
```

Comment:
👍  Actually since a Sudoku board is always 9x9, scaling never changes so it's O(1) instead of O(n^2)